src/ILCompiler.ReadyToRun/src/IBC/IBCProfileParser.cs

```diff
@@ -0,0 +1,785 @@
+                                    // contained no such methods prior to .NET 4.0.
+                                    //
+                                    // In all other cases, ZapSig::GetSignatureForTypeHandle writes the owning type signature
+                                    // in a "standard" form which always starts with bytes that match the following grammar:
+                                    //
+                                    //      [ELEMENT_TYPE_MODULE_ZAPSIG {Index}] [ELEMENT_TYPE_GENERICINST] ELEMENT_TYPE_(CLASS|VALUETYPE) {Token}
+                                    //
+                                    // IBCMerge only supports the standard form.  Specifically, if the flattened form is
+                                    // ever processed, updateParamSig and remapParamSig are unable to reliably determine
+                                    // whether the subsequent method token should be interpreted as a MethodDef or as an
+                                    // ibcMethodSpec (because the flattened form "hides" the explicit ELEMENT_TYPE_MODULE_ZAPSIG
+                                    // byte that is normally used to determine how to interpret the method token).
+                                    //
+                                    // Probe the leading bytes of the signature and ignore this ParamMethodSpec if the leading
+                                    // bytes prove that the owning type signature was NOT encoded in the standard form.
+                                    //
+
+                                    byte[] signature = ((BlobEntry.SignatureEntry)blob).Signature;
+                                    bool owningTypeSignatureAppearsToBeEncodedInStandardForm = false;
+
+                                    if (signature.Length >= 2)
+                                    {
+                                        CorElementType leadingByte = (CorElementType)signature[0];
+
+                                        if ((leadingByte == CorElementType.ELEMENT_TYPE_MODULE_ZAPSIG) ||
+                                            (leadingByte == CorElementType.ELEMENT_TYPE_GENERICINST) ||
+                                            (leadingByte == CorElementType.ELEMENT_TYPE_CLASS) ||
+                                            (leadingByte == CorElementType.ELEMENT_TYPE_VALUETYPE))
+                                        {
+                                            owningTypeSignatureAppearsToBeEncodedInStandardForm = true;
+                                        }
+                                    }
+
+                                    if (owningTypeSignatureAppearsToBeEncodedInStandardForm)
+                                    {
+                                        //                                        UTILS.CheckAndSetHighest(ref result.highestIbcMethodSpecToken, blob.Token);
```

Comment:
What's this?

Reply:
Copy paste issue with code that doesn't make sense in this codebase.